src/utils/event_loop.c

```diff
@@ -617,11 +617,15 @@ event_loop_unregister_event(struct event_loop *loop, int fd)
  * loop iteration.
  *
  * @param loop The event loop instance.
+ * @param retval The return value, 0 if success, an error code otherwise
+ * @return 0 if it exited cleanly, an error code otherwise
  */
-void
-event_loop_stop(struct event_loop *loop)
+int
+event_loop_stop(struct event_loop *loop, int retval)
```

Comment:
nit: might as well name this `exit_code` to match up with the contextual variable name.